### ISSUE TITLE
Broken features (no hover information, autocomplete, etc.) in my project's folder structure 

### ISSUE THREAD

**Issue by Raxelgrande** [scalameta/metals#6280]
### Describe the bug

During the development of my project, i did a code restructuring (from a single main file, to 3, with shared libraries between 2 of those files) and Metals started behaving strangely.
![Captura de pantalla_20240403_222920](https://github.com/scalameta/metals/assets/93939943/5d223d3e-7aff-4f38-a0db-46b838457feb)
After the restructuring, the following features stopped working:
- Goto definition (for user created functions)
- Autocompletion (for everything except that it does show already used words but not for example completing a string with the string compatible functions)
- Hover (broken for user created functions)

Unfortunately i didn't get anything back from Metals detecting that something was wrong. I actually got very surprised when i discovered that it kept working. 
When i created misc.scala , i only used imports from Scala's stdlib and all of Metals' features worked like they should. 

My guess is that this problem has to do with how i structured the directories in my project or maybe with how i used the import/package features. 

Example of how i am importing things:
```scala
package iris.interface
import bananatui.*
import iris.config._
import iris.theming._
import iris.themeSelector._
```
You can get my affected project, Iris, for testing, [here](https://github.com/Raxelgrande/Iris)

### Operating system

Linux

### Editor/Extension

VS Code

### Version of Metals

v1.29.0

**Comment by tgodzik** [comment]
Thanks for reporting! I see that you don't have any build tool in your workspace, which means Metals doesn't exactly know anything about your project structure. In that case we could (and should) startup [scala-cli](https://scala-cli.virtuslab.org/) or you can do it manually via `scala-cli setup-ide .`. However, It seems you have two mains in your project that have empty package, which means we will not be able to compile it. If you add `package something;` to one of the mains things should work with scala-cli.

I will see if we can start it manually.

**Comment by tgodzik** [comment]
Ach ok, so we try to detect if something is a Scala project, but only using specific directories. I will see if it makes sense to change that.

**Comment by tgodzik** [comment]
As an added bonus if you use Scala CLI you can try creating native image or Scala native applications, which don't require Java to run.

**Comment by tgodzik** [comment]
Once the release job is done you can try the latest snapshot as showed in https://scalameta.org/metals/docs/

IT should start scala-cli for you automatically, otherwise if you are wary of using a snapshot version you can install scala-cli, which will be the oficiall Scala runner

**Comment by Raxelgrande** [comment]
I tested the merged fixes using Metals 1.2.2+117-8cc7a42f-SNAPSHOT, all of Metals' features work properly now. 

But i found another issue, this message appears every time i start VSCode:
![Captura de pantalla_20240406_210312](https://github.com/scalameta/metals/assets/93939943/0cb457ec-bf5d-477a-81bc-08538c766fc7)
This is what appears on Metals Doctor:
![Captura de pantalla_20240406_211647](https://github.com/scalameta/metals/assets/93939943/c2615262-68d0-440e-baa7-d3e2b3a5a597)

**Comment by tgodzik** [comment]
Looks like we cannot find JDK sources, so no navigation or docs will be available, but you can ignore it if that doesn't matter to you. The expected paths for sources are in the recommendation. Unfortunately, there is no sensible way of downloading them standalone.
